networkx/algorithms/tests/test_swap.py

```diff
@@ -12,13 +12,29 @@ def test_directed_edge_swap():
     assert out_degrees == sorted((n, d) for n, d in G.out_degree())
 
 
+def test_edge_cases_directed_edge_swap():
+    graph = nx.path_graph(4, create_using=nx.DiGraph)
+    with pytest.raises(nx.NetworkXAlgorithmError):
```

Comment:
Generally looks good - let's add a `match=` to the `raises` context managers to ensure we are indeed catching the exceptions we expect for these cases.